test/dict.jl

```diff
@@ -830,6 +830,21 @@ Dict(1 => rand(2,3), 'c' => "asdf") # just make sure this does not trigger a dep
 
     @test_throws ArgumentError WeakKeyDict([1, 2, 3])
 
+    # WeakKeyDict does not convert keys
+    @test_throws ArgumentError WeakKeyDict{Int,Any}(5.0=>1)
+
+    # WeakKeyDict hashes with object-id
+    AA = copy(A)
```

Comment:
``julia
GC.@preserve A AA begin
...
end
```

Reply:
Thanks, fixed! Quesion: why would they need to be preserved?  The do not go out of scope and thus should not be GC'ed, no?

Reply:
`A` is used subsequently so is relatively safe but `AA` isn't so it could be garbage collected.